### ISSUE TITLE
make graphAPI endpoints customizable

### ISSUE THREAD

**Issue by davidcoutadeur** [lsc-project/lsc-microsoft-graph-api-plugin#3]
My customer has multiple instance of graphAPI.
One of them is hosted in China and so has customized endpoints.
It would be nice to have these parameters customized in lsc.xml

I have listed these urls (may be non exhaustive):

in MsGraphApiAuthentication.java, the authentication URL:
```
        WebTarget authTarget = ClientBuilder.newClient()
            .register(JacksonFeature.class)
            .target("https://login.microsoftonline.com/")
            .path(tenant)
            .path("oauth2/v2.0/token");
```

in MsGraphApiDao.java, the users endpoint:
```
        usersClient = client
            .target("https://graph.microsoft.com")
            .path("v1.0")
            .path(USER_PATH);
```

For the moment, I only consider the need to have the FQDN customizable.

**Comment by davidcoutadeur** [comment]
@remk @rouazana could you review my patch for the configuration parameters?
(for info, I haven't tested yet, it just passes the compilation)

**Comment by remk** [comment]
:+1: it looks good to me

**Comment by davidcoutadeur** [comment]
With the last patch, the code is functional on a customized tenant.

**Comment by rouazana** [comment]
> With the last patch, the code is functional on a customized tenant.

can you explain this fix? Anyway it could probably be rewritten without regxep

**Comment by davidcoutadeur** [comment]
the default hard-coded scope during authentication was `https://graph.microsoft.com/.default`

However, this URL depends of the usersURL parameter. I made it built on the top of the build URL.

I didn't find documentation explaining if there could be other type of scopes. If so, maybe we would have to make it a completely distinct parameter.

**Comment by rouazana** [comment]
thank you, I then prefer than we have a completely different scope parameter, it will make the intent clearer (and avoid this regexp ^^)

**Comment by davidcoutadeur** [comment]
regexes are beautiful :)

Last patch create  a new parameter for the scope. I have tested right now, it's working on my use case. (with customized endpoints and standard endpoints)

**Comment by rouazana** [comment]
thank you, that's far better. We could avoid variable re-allocation, but it's not a big issue for me in this case.

Can you open a pull request so that we can merge it?

**Comment by davidcoutadeur** [comment]
The new parameters:  `authenticationURL`, `usersURL` and `scope` were mandatory.
I have made them optional.

With this last fix, I think the feature is ready. I am making the MR

**Comment by rouazana** [comment]
fixed in v1.1
